### ISSUE TITLE
BUG: Build failed ->  TypeError: cannot inherit frozen dataclass from a non-frozen one

### ISSUE THREAD

**Issue by mrJakez** [stravalib/stravalib#596]
### Stravalib version checks

- [X] I have tested this in a new clean environment with only stravalib and core python files.

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/stravalib/) of stravalib.

- [X] I have confirmed this bug exists on the [main branch](https://github.com/stravalib/stravalib) of stravalib.

### What operating system are you seeing the problem on?

Linux

### What version of python or you running?

```python
3.13
```

### Reproducible Example

```python
from stravalib import Client
```

### Issue Description

I use a project with the modules stravalib, open, sqlmodel and fastapi. As soon as I import the Client from the stravalib package I get the following issue: 

> [+] Running 0/0
> 
>  ⠋ Container vestaboard_control-api-1  Recreated                                                                                                                                                                 0.1s 
> Attaching to api-1
> api-1  | INFO:     Started server process [1]
> api-1  | INFO:     Waiting for application startup.
> api-1  | INFO:     Application startup complete.
> api-1  | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
> api-1  | log me biatsch
> api-1  | INFO:     192.168.65.1:49580 - "GET / HTTP/1.1" 200 OK
> api-1  | INFO:     Shutting down
> api-1  | INFO:     Waiting for application shutdown.
> api-1  | INFO:     Application shutdown complete.
> api-1  | INFO:     Finished server process [1]
> api-1  | 
> api-1  | Aborted!
> api-1 exited with code 0
> api-1 has been recreated
> api-1  | Traceback (most recent call last):
> api-1  |   File "/usr/local/bin/uvicorn", line 8, in <module>
> api-1  |     sys.exit(main())
> api-1  |              ~~~~^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/click/core.py", line 1130, in __call__
> api-1  |     return self.main(*args, **kwargs)
> api-1  |            ~~~~~~~~~^^^^^^^^^^^^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/click/core.py", line 1055, in main
> api-1  |     rv = self.invoke(ctx)
> api-1  |   File "/usr/local/lib/python3.13/site-packages/click/core.py", line 1404, in invoke
> api-1  |     return ctx.invoke(self.callback, **ctx.params)
> api-1  |            ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/click/core.py", line 760, in invoke
> api-1  |     return __callback(*args, **kwargs)
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/main.py", line 410, in main
> api-1  |     run(
> api-1  |     ~~~^
> api-1  |         app,
> api-1  |         ^^^^
> api-1  |     ...<45 lines>...
> api-1  |         h11_max_incomplete_event_size=h11_max_incomplete_event_size,
> api-1  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> api-1  |     )
> api-1  |     ^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/main.py", line 577, in run
> api-1  |     server.run()
> api-1  |     ~~~~~~~~~~^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/server.py", line 65, in run
> api-1  |     return asyncio.run(self.serve(sockets=sockets))
> api-1  |            ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/asyncio/runners.py", line 194, in run
> api-1  |     return runner.run(main)
> api-1  |            ~~~~~~~~~~^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/asyncio/runners.py", line 118, in run
> api-1  |     return self._loop.run_until_complete(task)
> api-1  |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/asyncio/base_events.py", line 721, in run_until_complete
> api-1  |     return future.result()
> api-1  |            ~~~~~~~~~~~~~^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/server.py", line 69, in serve
> api-1  |     await self._serve(sockets)
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/server.py", line 76, in _serve
> api-1  |     config.load()
> api-1  |     ~~~~~~~~~~~^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/config.py", line 434, in load
> api-1  |     self.loaded_app = import_from_string(self.app)
> api-1  |                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/site-packages/uvicorn/importer.py", line 19, in import_from_string
> api-1  |     module = importlib.import_module(module_str)
> api-1  |   File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
> api-1  |     return _bootstrap._gcd_import(name[level:], package, level)
> api-1  |            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> api-1  |   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
> api-1  |   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
> api-1  |   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
> api-1  |   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
> api-1  |   File "<frozen importlib._bootstrap_external>", line 1022, in exec_module
> api-1  |   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
> api-1  |   File "/app/main.py", line 5, in <module>
> api-1  |     from stravalib import Client
> api-1  |   File "/usr/local/lib/python3.13/site-packages/stravalib/__init__.py", line 1, in <module>
> api-1  |     from stravalib.client import Client
> api-1  |   File "/usr/local/lib/python3.13/site-packages/stravalib/client.py", line 30, in <module>
> api-1  |     import pint
> api-1  |   File "/usr/local/lib/python3.13/site-packages/pint/__init__.py", line 18, in <module>
> api-1  |     from .delegates.formatter._format_helpers import formatter
> api-1  |   File "/usr/local/lib/python3.13/site-packages/pint/delegates/__init__.py", line 12, in <module>
> api-1  |     from . import txt_defparser
> api-1  |   File "/usr/local/lib/python3.13/site-packages/pint/delegates/txt_defparser/__init__.py", line 12, in <module>
> api-1  |     from .defparser import DefParser
> api-1  |   File "/usr/local/lib/python3.13/site-packages/pint/delegates/txt_defparser/defparser.py", line 10, in <module>
> api-1  |     from . import block, common, context, defaults, group, plain, system
> api-1  |   File "/usr/local/lib/python3.13/site-packages/pint/delegates/txt_defparser/common.py", line 23, in <module>
> api-1  |     @dataclass(frozen=True)
> api-1  |      ~~~~~~~~~^^^^^^^^^^^^^
> api-1  |   File "/usr/local/lib/python3.13/dataclasses.py", line 1295, in wrap
> api-1  |     return _process_class(cls, init, repr, eq, order, unsafe_hash,
> api-1  |                           frozen, match_args, kw_only, slots,
> api-1  |                           weakref_slot)
> api-1  |   File "/usr/local/lib/python3.13/dataclasses.py", line 1043, in _process_class
> api-1  |     raise TypeError('cannot inherit frozen dataclass from a '
> api-1  |                     'non-frozen one')
> api-1  | TypeError: cannot inherit frozen dataclass from a non-frozen one

As soon as I remove `from stravalib import Client` from my python script it works. Any suggestions where I can look at to fix this issue? I'm a bit lost. I fear it is related to the pydantic dependency. Any advices?

Cheers
Dennis

### Expected Behavior

That I can include the stravalib library :)

### Your environment

<details>

/app # pip freeze
anyio==4.4.0
appdirs==1.4.4
APScheduler==3.10.4
arrow==1.3.0
blinker==1.8.1
certifi==2024.2.2
charset-normalizer==3.3.2
click==8.1.3
debugpy==1.8.1
distro==1.9.0
exceptiongroup==1.2.1
fastapi==0.115.4
Flask==3.0.3
Flask-APScheduler==1.13.1
flexcache==0.3
flexparser==0.3.1
h11==0.14.0
httpcore==1.0.5
httpx==0.27.0
icalendar==5.0.13
idna==3.6
importlib-metadata==5.1.0
itsdangerous==2.1.2
Jinja2==3.1.2
jiter==0.7.0
MarkupSafe==2.1.1
openai==1.53.0
Pint==0.24.3
pydantic==1.10.18
python-dateutil==2.9.0.post0
pytz==2024.1
requests==2.31.0
six==1.16.0
sniffio==1.3.1
SQLAlchemy==2.0.36
sqlmodel==0.0.22
starlette==0.41.2
stravalib==1.7
tqdm==4.66.6
types-python-dateutil==2.9.0.20241003
typing_extensions==4.12.0
tzlocal==5.2
urllib3==2.2.1
uvicorn==0.30.1
vesta==0.12.0
Werkzeug==3.0.2
zipp==3.11.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lwasser** [comment]
hey @mrJakez 👋🏻 
Welcome to stravalib! Can you please try to reinstall / recreate your environment using the most current version of stravalib?

I see you checked that you are using the most current version however in your environment i see this:

```
pydantic==1.10.18
python-dateutil==2.9.0.post0
pytz==2024.1
requests==2.31.0
six==1.16.0
sniffio==1.3.1
SQLAlchemy==2.0.36
sqlmodel==0.0.22
starlette==0.41.2
stravalib==1.7
```

The most current version of stravalib is 2.x and pydantic should also be 2.x rather than 1.x. Stravalib 2.x has a lot of upgrades and fixes.

**Comment by Apocrathia** [comment]
It looks like this may stem from a bug in the recent version of `pint`. https://github.com/hgrecco/pint/issues/1969

**Comment by mrJakez** [comment]
@lwasser : Thanks for your feedback! I started to build a fresh project to set this up, but thanks to @Apocrathia, the solution is much easier - I just downgraded the base image to python:3.12-alpine > now everything is working as expected! I will monitor the issue to check when I can switch back to 3.13.

Thanks a lot to both of you for your quick help and feedback! - I will close this issue here right now as it is obviously not related to stravalib. ;)
